dev-tool/src/list_net/inspect.rs

```diff
@@ -0,0 +1,662 @@
+            )
+        };
+        Some(Ok((
+            format!("{:0>width$}.{}", idx, name, width = nr_digits),
+            array,
+        )))
+    });
+    Ok(Box::new(iter))
+}
+
+fn write_failure_message(
+    out: &mut String,
+    title: &str,
+    array: &[String],
+) -> Result<(), fmt::Error> {
+    use fmt::Write;
+    writeln!(out, " {}:", title)?;
+    for array_name in array {
+        writeln!(out, " - {}", array_name)?;
+    }
+    Ok(())
+}
+
+/// Parses the filter by splitting each filter sequence and trimming each filter.
+fn parse_filter(filter: Vec<String>) -> HashSet<String> {
+    filter
+        .iter()
+        .flat_map(|names| {
+            names
+                .split(',')
+                .map(|name| name.trim().to_owned())
+                .filter(|name| !name.is_empty())
+        })
+        .collect::<HashSet<_>>()
+}
+
+/// Parses a `<from>..=<to>` (f32) range.
+fn parse_range(range: impl AsRef<str>) -> Result<RangeInclusive<f32>, Error> {
+    let mut split = range.as_ref().split("..=");
+    let first = split.next().unwrap();
+    if let Some(second) = split.next() {
+        if split.next().is_some() {
+            bail!("Only <from>..=<to> syntax is currently allowed.");
+        }
+        let first: f32 = first.trim().parse()?;
+        let second: f32 = second.trim().parse()?;
+        Ok(first..=second)
+    } else {
+        bail!("Only <from>..=<to> syntax is currently allowed.");
+    }
+}
+
+/// Stats for a matrix.
+struct Stats {
+    /// The max element in the matrix, or 0 if the matrix is empty.
+    min: f32,
+    /// The min element in the matrix, or 0 if the matrix is empty.
+    max: f32,
+    /// The arithmetic mean of the elements in the matrix, or 0 if the matrix is empty.
+    mean: f32,
+    /// The standard derivation (ddof=0) of the elements in the matrix.
+    std: f32,
+    /// Is `true`, if the matrix contains `NaN` elements.
+    has_nans: bool,
+    /// Is `true`, if the matrix contains `Inf` elements (independent of sign).
+    has_infs: bool,
+    /// Is `true`, if the matrix contains subnormal elements.
+    has_subnormals: bool,
+    /// True if any element is not normal.
+    normal_checks_failed: bool,
+    /// True if any range checks failed.
+    range_checks_failed: bool,
+}
+
+impl Stats {
+    fn calculate<S, D>(
+        array: ArrayBase<S, D>,
+        do_normal_checks: bool,
+        range_checks: Option<&RangeInclusive<f32>>,
+    ) -> Self
+    where
+        S: Data<Elem = f32>,
+        D: Dimension,
+    {
+        array.iter().copied().fold(
+            Self {
+                min: 0.0,
+                max: 0.0,
+                mean: array.mean().unwrap_or_default(),
+                std: array.std(0.),
+                has_nans: false,
+                has_infs: false,
+                has_subnormals: false,
+                normal_checks_failed: false,
+                range_checks_failed: false,
+            },
+            |mut self_, val| {
```

Comment:
this is probably a matter of preference, i would usually avoid trailing underscores and name the variable sth like `this` or in this case wrt to the type `stats`.